File: web/src/types/specimen.ts
import type { ObservationRequest } from "./observation_requests";

export interface Specimen {
    id:                   number;
    specimen_hl7_id:      string;
    patient_id:           number;
    order_id:             number;
    type:                 string;
    collection_date:      string;
    received_date:        string;
    source:               string;
    condition:            string;
    method:               string;
    comments:             string;
    barcode:              string;
    created_at:           string;
    updated_at:           string;
    observation_result:   any[];
    observation_requests: ObservationRequest[];
    test_result:          null;
}

File: integration-service/internal/models/specimen.go
package models

import "time"

type Specimen struct {
	ID             int
	PatientID      int
	OrderID        int
	Type           string
	CollectionDate string
	ReceivedDate   time.Time
	Source         string
	Condition      string
	Method         string
	Comments       string
	Barcode        string
	CreatedAt      time.Time
	UpdatedAt      time.Time

	// Relationships
	ObservationResult  []ObservationResult  `gorm:"foreignKey:SpecimenID"`
	ObservationRequest []ObservationRequest `gorm:"foreignKey:SpecimenID;->"`
	WorkOrder          WorkOrder            `gorm:"foreignKey:OrderID;->"`
	Patient            Patient              `gorm:"foreignKey:PatientID;->"`
}
